mlflow/recipes/steps/train.py

```diff
@@ -327,40 +335,69 @@ def my_warn(*args, **kwargs):
+
+                model_uri = get_step_output_path(
                     recipe_root_path=self.recipe_root,
                     step_name=self.name,
                     relative_path=TrainStep.MODEL_ARTIFACT_RELATIVE_PATH,
                 )
-                if os.path.exists(output_model_path) and os.path.isdir(output_model_path):
-                    shutil.rmtree(output_model_path)
-
-                if self.predict_proba:
-                    mlflow.sklearn.save_model(
-                        model, output_model_path, pyfunc_predict_fn="predict_proba"
+                sklearn_model_uri = get_step_output_path(
+                    recipe_root_path=self.recipe_root,
+                    step_name=self.name,
+                    relative_path=TrainStep.SKLEARN_MODEL_ARTIFACT_RELATIVE_PATH,
+                )
+                if os.path.exists(model_uri) and os.path.isdir(model_uri):
+                    shutil.rmtree(model_uri)
+                if os.path.exists(sklearn_model_uri) and os.path.isdir(sklearn_model_uri):
+                    shutil.rmtree(sklearn_model_uri)
+
+                # Saving the sklearn model along with its pyfunc model since mlflow evaluate
+                # predicts results using an sklearn model
+                mlflow.sklearn.save_model(trained_pipeline, sklearn_model_uri)
+                artifacts = {"model_path": sklearn_model_uri}
+                with TempDir() as tmp:
+                    # Saving a temp model so that signature of the model output could be inferred
+                    # before the pyfunc model can be saved as the specified model_uri location
+                    pyfunc_model_tmp_path = os.path.join(tmp.path(), "pyfunc_model")
+                    mlflow.pyfunc.save_model(
+                        path=pyfunc_model_tmp_path,
+                        python_model=wrapped_model,
+                        artifacts=artifacts,
                     )
-                else:
-                    mlflow.sklearn.save_model(model, output_model_path)
+                    tempModel = mlflow.pyfunc.load_model(pyfunc_model_tmp_path)
+                    model_schema = infer_signature(
+                        raw_X_train, tempModel.predict(raw_X_train.copy())
+                    )
+                    mlflow.pyfunc.save_model(
+                        path=model_uri,
+                        python_model=wrapped_model,
+                        artifacts=artifacts,
+                        signature=model_schema,
+                        code_path=self.code_paths,
+                    )
+                model = mlflow.pyfunc.load_model(model_uri)
+                model_info = Model.load(model_uri)
+                model_data_subpath = os.path.join(
+                    "artifacts", TrainStep.SKLEARN_MODEL_ARTIFACT_RELATIVE_PATH, "model.pkl"
+                )
+                # Adding a sklearn flavor to the pyfunc model so models could be loaded easily
+                # using mlflow.sklearn.load_model
```

Comment:
Can we move this comment up above line 385? `model_info = Model.load(model_uri)`